pkg/rbd/nodeserver.go

```diff
@@ -303,15 +303,37 @@ func (ns *NodeServer) mountVolumeToStagePath(ctx context.Context, req *csi.NodeS
 	// Publish Path
 	fsType := req.GetVolumeCapability().GetMount().GetFsType()
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
+	source := devicePath
+	readOnly := false
+	existingFormat, err := diskMounter.GetDiskFormat(source)
+	if err != nil {
+		return err
+	}
+	if existingFormat == "" && !readOnly {
```

Comment:
Replace this with "if existingFormat == "" && (fsType == "ext4" || fsType == "xfs")  {".